wordpress.yaml

```diff
@@ -1,7 +1,158 @@
+      ufw:
+        rule: allow
+        name: "Nginx Full"
+        
+    - name: Configure SSH firewall
+      ufw:
+        rule: allow
+        name: "OpenSSH"
+      notify: Reload firewall
+
+    - name: Enable firewall
+      ufw:
+        state: enabled
+    
+    - name: Configure Nginx
+      template:
+        src: "/vagrant/nginx.conf.j2"
+        dest: "/etc/nginx/sites-available/{{ wp_conf_dir }}"
+
+    - name: Enable Nginx configuration
+      file:
+        src: "/etc/nginx/sites-available/{{ wp_conf_dir }}"  
+        dest: "/etc/nginx/sites-enabled/{{ wp_conf_dir }}"
+        state: link
+
+    - name: Unlink default Nginx
+      file:
+        path: "/etc/nginx/sites-enabled/default"
+        state: absent
+      notify: Restart Nginx  
+  
+    - name: Set root password in MYSQL
+      mysql_user:
+        name: root
+        password: "{{ mysql_password }}"
+        login_user: root
+        login_password: "{{ mysql_password }}"
+        login_unix_socket: "/var/run/mysqld/mysqld.sock"
+
+    - name: Remove all MYSQL anonymous users
+      mysql_user:
+        name: ''
+        host_all: yes
+        state: absent
+        login_user: root
+        login_password: "{{ mysql_password }}"
+  
+    - name: Remove MySQL test database
+      mysql_db:
+        name: test
+        state: absent
+        login_user: root
+        login_password: "{{ mysql_password }}"
+
+    - name: Create new database
+      mysql_db:
+        name: "wordpress"
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+
+    - name: Create WordPress user
+      mysql_user:
+        name: "wordpressuser"
+        password: "{{ wp_database_password }}"
+        priv: "*.*:ALL,GRANT"
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+
+    - name: Create directory
+      file:
+        path: "/var/www/{{ wp_conf_dir }}"
+        state: directory
+
+    - name: Download WordPress zip
+      get_url:
+        url: "https://wordpress.org/{{ wp_file }}"
+        dest: "/tmp/{{ wp_file }}"
+
+    - name: Check for extracted WordPress
+      stat: 
+        path: "/var/www/{{ wp_conf_dir }}/wp-login.php"
+      register: extracted
+
+    - name: Extract WordPress
+      unarchive:
+        src: "/tmp/{{ wp_file }}"
+        dest: "/var/www/"
+      when: not extracted.stat.exists
+        
+    - name: Copy wp-config file to wordpress directory
+      copy:
+        src: "/vagrant/wp-config.php"
+        dest: "/var/www/{{ wp_conf_dir }}/wp-config.php"
+        
+    - name: Install WP-CLI
+      get_url:
+        url: "{{ wp_cli_url }}"
+        dest: "/usr/local/bin/wp"
+        mode: 0755
+
+    - name: Change directory to /var/www/wordpress and install WP
+      command: wp core install --url="192.168.50.2" --title="WordPress Setup" --admin_user=admin --admin_email=admin@test.com --admin_password="!2three456." --path=/var/www/wordpress --allow-root
```

Comment:
not a mistake, but you have variable with `192.168.50.2` why not using it?

BTW, also as you did with MySQL credentials, maybe it make sence to use Wordpress configurations from `variables.yaml` file?